iceoryx_hoofs/source/posix_wrapper/unix_domain_socket.cpp

```diff
@@ -178,7 +179,16 @@ expected<IpcChannelError> UnixDomainSocket::closeFileDescriptor() noexcept
         {
             if (IpcChannelSide::SERVER == m_channelSide)
             {
-                unlink(&(m_sockAddr.sun_path[0]));
+                auto unlinkCall = posixCall(unlink)(&(m_sockAddr.sun_path[0]))
+                                      .failureReturnValue(ERROR_CODE)
+                                      .ignoreErrnos(ENOENT)
+                                      .evaluate();
+                if (unlinkCall.has_error())
+                {
+                    IOX_LOG(ERROR) << "Unable to close file descriptor " << &(m_sockAddr.sun_path[0])
+                                   << "task failed with error: " << unlinkCall->getHumanReadableErrnum();
```

Comment:
@zmostafa 

This is not required, when you use `posixCall` it will generate a detailed error log message for you stating the file, line and function as well as the errno code and a human readable version of the error.

On this level you provide now context where this error has happened.

So in this case.
1. `posixCall` states that `unlink` has failed and provides all the little details to it
2. on this level you would state the the IpcChannel could not be cleaned up due to a failure in unlink
3. A higher level may inform the user that roudi could not start up since there were resources remaining in the system which could not be cleaned up.

If then this failure would occur the user gets a nice detailed log on what happened and what caused it.